### PR TITLE
update to gradle 7

### PR DESCRIPTION
**Pull request by cdietrich**
update to gradle 7
see https://github.com/eclipse/xtext/issues/1548

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
i still see problem with the double acting
of core.tests as test and lib project
somehow stuff is not visible to java8 tests in eclipse

**Comment by cdietrich** [comment]
Screenshot 2021-08-12 142822

**Comment by cdietrich** [comment]
so the question is:
how can i have core.tests as api and tests. (reexport the tests)

File: org.eclipse.xtend.core.tests.java8/build.gradle [hunk 1]
```diff
@@ -1,5 +1,6 @@
 dependencies {
-	compile project(':org.eclipse.xtend.core.tests')
+	testImplementation project(':org.eclipse.xtend.core.tests')
+	testImplementation project(':org.eclipse.xtend.core.tests').sourceSets.test.output // TODO is there a better way
```
Comment (oehme) [inline org.eclipse.xtend.core.tests.java8/build.gradle:3]: Use the [test-fixtures](https://docs.gradle.org/current/userguide/java_testing.html#sec:java_test_fixtures) plugin to share the test fixtures from one project with another.

Comment (cdietrich) [reply org.eclipse.xtend.core.tests.java8/build.gradle]: seems to work. thanks.

Comment (cdietrich) [reply org.eclipse.xtend.core.tests.java8/build.gradle:3]: hmm not quite yet. will create follow up issue

**Review by szarnekow — APPROVED** [review]

**Pull request merged**